src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -207,52 +238,209 @@ public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context,
             return CommonErrors.NotYetImplementedError(irContext, $"Text format for {args[0]?.GetType().Name}");
         }
 
-        internal static string ExpandDateTimeFormatSpecifiers(string format, CultureInfo culture)
+        internal static string ExpandDateTimeExcelFormatSpecifiers(EvalVisitor runner, string format, string defaultFormat, DateTime dateTime, CultureInfo culture)
```

Comment:
>ExpandDateTimeExcelFormatSpecifiers [](http://example.com/codeflow?start=31&length=35)

This code is all ported from JS? 
Ie, so same algorithm we're using in Power Apps client-side?

Reply:
Only the code that handles custom formats were ported from JS. All JS tests have been successful. @CarlosFigueiraMSFT please double-check if there is anything you would change.